File: ThreeWayPartition/ThreeWayPartition.py
def ThreeWayPartition(arr, n, lowVal, highVal):
    start = 0
    end = n - 1
    i = 0
    while i <= end:
        if arr[i] < lowVal:
            temp = arr[i]
            arr[i] = arr[start]
            arr[start] = temp
            i += 1
            start += 1
        elif arr[i] > highVal:
            temp = arr[i]
            arr[i] = arr[end]
            arr[end] = temp
            end -= 1
 
        else:
            i += 1
 
# Driver code
if __name__ == "__main__":
    arr = [1, 14, 5, 20, 4, 2, 54,
           20, 87, 98, 3, 1, 32]
    n = len(arr)
 
    ThreeWayPartition(arr, n, 10, 20)
 
    print(arr)

File: ThreeWayPartition/ThreeWayPartition.js
function ThreeWayPartition(arr, n, lowVal, highVal) {
  let start = 0,
    end = n - 1;
  for (let i = 0; i <= end; ) {
    if (arr[i] < lowVal) {
      let temp = arr[start];
      arr[start] = arr[i];
      arr[i] = temp;
      start++;
      i++;
    } else if (arr[i] > highVal) {
      let temp = arr[end];
      arr[end] = arr[i];
      arr[i] = temp;
      end--;
    } else i++;
  }
}

// Driver code
let arr = [1, 14, 5, 20, 4, 2, 54, 20, 87, 98, 3, 1, 32];
let n = arr.length;

ThreeWayPartition(arr, n, 10, 20);

console.log(arr);
